sdk/resource/detect/auto.go

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
Can we move this to the `resource` package itself to keep the context of it's functionality in the usage (e.g. `resource.Detect`) and to match the OTEP closer.